File: GFG/LinkedList/revdl.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

void reversedl(DLList<int> &dl){
    if(dl.head==nullptr){
        return;
    }
    DNode<int>*curr=dl.head;
    DNode<int>* pre=NULL;
    while(curr!=nullptr){
        pre=curr->prev;
        curr->prev=curr->next;
        curr->next=pre;
        curr=curr->prev;
    }
    dl.head=pre->prev;
    return;
}
int main(){
    DLList<int> dll;
    dll.insert(5);
    dll.insert(10);
    dll.insert(15);
    dll.insert(20);
    dll.display();
    reversedl(dll);
    dll.display();
    return 0;
}

File: GFG/LinkedList/revdl.py
import ds

def reversedll(dl):
    if(dl.head is None):
        return
    curr=dl.head
    while(curr.next!=None):
        curr.next,curr.prev=curr.prev,curr.next
        curr=curr.prev
    dl.head=curr
    temp=curr.next
    curr.next=curr.prev
    curr.prev=temp
    
dll=ds.DLList()
dll.insert(5)
dll.insert(10)
dll.insert(15)
dll.insert(20)
dll.display()
reversedll(dll)
dll.display()
